File: myExpenses/src/androidTest/java/org/totschnig/myexpenses/test/provider/CurrencyTest.java
package org.totschnig.myexpenses.test.provider;

import org.totschnig.myexpenses.model.AccountType;
import org.totschnig.myexpenses.provider.DatabaseConstants;
import org.totschnig.myexpenses.provider.MoreDbUtilsKt;
import org.totschnig.myexpenses.provider.TransactionProvider;
import org.totschnig.myexpenses.testutils.BaseDbTest;

public class CurrencyTest extends BaseDbTest {

  private final CurrencyInfo TEST_CURRENCY = new CurrencyInfo("Bitcoin", "BTC");
  private final AccountInfo TEST_ACCOUNT = new AccountInfo("Account 0", AccountType.CASH, 0, TEST_CURRENCY.code);

  public void testShouldNotDeleteFrameworkCurrency() {
    try {
      getMockContentResolver().delete(TransactionProvider.CURRENCIES_URI.buildUpon().appendPath("EUR").build(), null, null);
      fail("Expected deletion of framework currency to fail");
    } catch (IllegalArgumentException e) {
      e.printStackTrace();
    }
  }

  public void testShouldDeleteUnUsedCurrency() {
    MoreDbUtilsKt.insert(mDb,
        DatabaseConstants.TABLE_CURRENCIES,
        TEST_CURRENCY.getContentValues()
    );
    int result = getMockContentResolver().delete(TransactionProvider.CURRENCIES_URI.buildUpon().appendPath(TEST_CURRENCY.code).build(), null, null);
    assertEquals(1, result);
  }

  public void testShouldNotDeleteUsedCurrency() {
    MoreDbUtilsKt.insert(mDb,
        DatabaseConstants.TABLE_CURRENCIES,
        TEST_CURRENCY.getContentValues()
    );
    MoreDbUtilsKt.insert(mDb,
        DatabaseConstants.TABLE_ACCOUNTS,
        TEST_ACCOUNT.getContentValues()
    );
    int result = getMockContentResolver().delete(TransactionProvider.CURRENCIES_URI.buildUpon().appendPath(TEST_CURRENCY.code).build(), null, null);
    assertEquals(0, result);
  }
}

File: myExpenses/src/androidTest/java/org/totschnig/myexpenses/test/provider/CurrencyTest.kt
package org.totschnig.myexpenses.test.provider

import junit.framework.TestCase
import org.totschnig.myexpenses.provider.AccountInfo
import org.totschnig.myexpenses.provider.DatabaseConstants
import org.totschnig.myexpenses.provider.TransactionProvider
import org.totschnig.myexpenses.provider.insert
import org.totschnig.myexpenses.testutils.BaseDbTest
import org.totschnig.myexpenses.testutils.CurrencyInfo

class CurrencyTest : BaseDbTest() {
    private val testCurrency = CurrencyInfo("Bitcoin", "BTC")

    fun testShouldNotDeleteFrameworkCurrency() {
        try {
            getMockContentResolver().delete(
                TransactionProvider.CURRENCIES_URI.buildUpon().appendPath("EUR").build(), null, null
            )
            fail("Expected deletion of framework currency to fail")
        } catch (e: IllegalArgumentException) {
            e.printStackTrace()
        }
    }

    fun testShouldDeleteUnUsedCurrency() {
        mDb
            .insert(
                DatabaseConstants.TABLE_CURRENCIES,
                testCurrency.contentValues
            )
        val result = getMockContentResolver().delete(
            TransactionProvider.CURRENCIES_URI.buildUpon().appendPath(testCurrency.code).build(),
            null,
            null
        )
        TestCase.assertEquals(1, result)
    }

    fun testShouldNotDeleteUsedCurrency() {
        val testAccount = AccountInfo("Account 0", cashAccount.id, 0, testCurrency.code)

        mDb
            .insert(
                DatabaseConstants.TABLE_CURRENCIES,
                testCurrency.contentValues
            )
        mDb
            .insert(
                DatabaseConstants.TABLE_ACCOUNTS,
                testAccount.contentValues
            )
        val result = getMockContentResolver().delete(
            TransactionProvider.CURRENCIES_URI.buildUpon().appendPath(testCurrency.code).build(),
            null,
            null
        )
        TestCase.assertEquals(0, result)
    }
}
